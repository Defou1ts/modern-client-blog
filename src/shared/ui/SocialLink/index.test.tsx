import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import { SocialLink } from './index';

describe('SocialLink component', () => {
	it('renders a Facebook SocialLink', () => {
		render(<SocialLink type="facebook" path="/" />);
		const text = screen.getByAltText('Facebook');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a Instagram SocialLink', () => {
		render(<SocialLink type="instagram" path="/" />);
		const text = screen.getByAltText('Instagram');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a Twitter SocialLink', () => {
		render(<SocialLink type="twitter" path="/" />);
		const text = screen.getByAltText('Twitter');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a Linkedin SocialLink', () => {
		render(<SocialLink type="linkedin" path="/" />);
		const text = screen.getByAltText('Linkedin');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('facebook SocialLink snapshot', () => {
		const component = renderer.create(<SocialLink type="facebook" path="/" />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('instagram SocialLink snapshot', () => {
		const component = renderer.create(<SocialLink type="instagram" path="/" />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('twitter SocialLink snapshot', () => {
		const component = renderer.create(<SocialLink type="twitter" path="/" />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('linkedin SocialLink snapshot', () => {
		const component = renderer.create(<SocialLink type="linkedin" path="/" />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
