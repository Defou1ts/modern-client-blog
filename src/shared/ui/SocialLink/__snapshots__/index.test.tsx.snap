// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLink component facebook SocialLink snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
  target="_blank"
>
  <img
    alt="Facebook"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    title="Facebook"
    width={40}
  />
</a>
`;

exports[`SocialLink component instagram SocialLink snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
  target="_blank"
>
  <img
    alt="Instagram"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    title="Instagram"
    width={40}
  />
</a>
`;

exports[`SocialLink component linkedin SocialLink snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
  target="_blank"
>
  <img
    alt="Linkedin"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    title="Linkedin"
    width={40}
  />
</a>
`;

exports[`SocialLink component twitter SocialLink snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
  target="_blank"
>
  <img
    alt="Twitter"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    title="Twitter"
    width={40}
  />
</a>
`;
